netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -326,6 +327,28 @@ public NettyChannelBuilder keepAliveWithoutCalls(boolean enable) {
     return this;
   }
 
+
+  /**
+   * If non-{@code null}, attempts to create connections bound to a local port.
+   */
+  public NettyChannelBuilder localSocketPicker(@Nullable LocalSocketPicker localSocketPicker) {
+    this.localSocketPicker = localSocketPicker;
+    return this;
+  }
+
+  @ExperimentalApi("FIXME")
+  public interface LocalSocketPicker {
```

Comment:
I am not sure if this should be a transport-agnostic interface or netty-only, but I guess if we ever need to support it in other transports, we could just promote the setter to `ManagedChannelBuilder`.

Reply:
yeah, this is experimental.  Just trying it out.